javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -718,13 +725,13 @@ protected IType createSyncReturnWithResponseType(IType syncReturnType, Operation
     }
 
     protected ReturnValue createSimpleSyncRestResponseReturnValue(Operation operation, IType syncReturnWithResponse, IType syncReturnType) {
-        return new ReturnValue(returnTypeDescription(operation, syncReturnWithResponse,
-                syncReturnType), syncReturnWithResponse);
+        return new ReturnValue(returnTypeDescription(operation, syncReturnWithResponse, syncReturnType),
+                syncReturnWithResponse);
     }
 
-    protected ReturnValue createSimpleAsyncRestResponseReturnValue(Operation operation, ProxyMethod proxyMethod, IType syncReturnType) {
-        return new ReturnValue(returnTypeDescription(operation, proxyMethod.getReturnType().getClientType(), syncReturnType),
-                proxyMethod.getReturnType().getClientType());
+    protected ReturnValue createSimpleAsyncRestResponseReturnValue(Operation operation, IType asyncRestResponseReturnType, IType syncReturnType) {
+        return new ReturnValue(returnTypeDescription(operation, asyncRestResponseReturnType, syncReturnType),
+                asyncRestResponseReturnType);
     }
```

Comment:
It should be equivalent, as `asyncRestResponseReturnType` variable comes from `ProxyMethodMapper.getAsyncRestResponseReturnType`.